src/measure_theory/bochner_integration.lean

```diff
@@ -53,55 +516,129 @@ begin
 
-def to_simple_func (f : α →ₛ γ) : measure_theory.simple_func α γ := classical.some f.2
+variables (K : Type*) [normed_field K] [second_countable_topology K] [normed_space K β]
 
--- bochner integration over simple functions in l1 space
-def integral (f : α →ₛ γ) : γ :=
-(f.to_simple_func).range.sum (λ (x : γ), (ennreal.to_real (volume ((f.to_simple_func) ⁻¹' {x}))) • x)
+variables (α β)
 
-end integral
+def coe_to_l1 : (α →₁ₛ β) →L[K] (α →₁ β) :=
+{ to_fun := (coe : (α →₁ₛ β) → (α →₁ β)),
+  add := λf g, rfl,
+  smul := λk f, rfl,
+  cont := l1.simple_func.uniform_continuous.continuous, }
 
-end simple_func
+variables {α β K}
 
-open simple_func
+end coe_to_l1
 
-variables [normed_space ℝ γ]
+section simple_func_integral
 
--- bochner integration over functions in l1 space
-def integral (f : α →₁ γ) : γ :=
-dense_embedding_of_simple_func.to_dense_inducing.extend simple_func.integral f
+variables [normed_space ℝ β]
 
-end l1
+/-- Bochner integration over simple functions in l1 space. -/
+def integral (f : α →₁ₛ β) : β := simple_func.bintegral (f.to_simple_func)
 
-variables [normed_space ℝ γ]
+lemma integral_eq_lintegral {f : α →₁ₛ ℝ} (h_pos : ∀ₘ a, 0 ≤ f.to_simple_func a) :
+  integral f = ennreal.to_real (∫⁻ a, ennreal.of_real (f.to_simple_func a)) :=
+by { rw [integral, simple_func.bintegral_eq_lintegral' f.integrable h_pos], refl }
+
+lemma integral_congr (f g : α →₁ₛ β) (h : ∀ₘ a, f.to_simple_func a = g.to_simple_func a) :
+  integral f = integral g :=
+by { simp only [integral], apply simple_func.bintegral_congr f.integrable g.integrable, exact h }
 
-def integral (f : α → γ) : γ :=
-if hf : measurable f ∧ integrable f
-then (l1.mk f hf.1 hf.2).integral
-else 0
+lemma integral_add (f g : α →₁ₛ β) : integral (f + g) = integral f + integral g :=
+begin
+  simp only [integral],
+  rw ← simple_func.bintegral_add f.integrable g.integrable,
+  apply simple_func.bintegral_congr (f + g).integrable,
+    { exact integrable_add f.measurable g.measurable f.integrable g.integrable },
+    { apply add_to_simple_func },
+end
+
+lemma integral_smul (r : ℝ) (f : α →₁ₛ β) : integral (r • f) = r • integral f :=
+begin
+  simp only [integral],
+  rw ← simple_func.bintegral_smul _ f.integrable,
+  apply simple_func.bintegral_congr (r • f).integrable,
+    { exact integrable_smul _ f.integrable },
+    { apply smul_to_simple_func }
+end
+
+instance : is_add_group_hom (integral : (α →₁ₛ β) → β) := { map_add := integral_add }
+
+lemma tendsto_integral : tendsto (integral : (α →₁ₛ β) → β) (nhds 0) (nhds 0) :=
+begin
+  have := @metric.tendsto_nhds_nhds (α →₁ₛ β) β _ _ integral 0 0,
+  rw this,
+  assume ε ε0,
+  use ε, use ε0,
+  assume f,
+  rw [dist_zero_right, norm_eq_bintegral, dist_zero_right, integral],
+  assume hf,
+  exact lt_of_le_of_lt (f.to_simple_func.norm_bintegral_le_bintegral_norm f.integrable) hf
+end
+
+lemma uniform_continuous_integral : uniform_continuous (integral : (α →₁ₛ β) → β) :=
+uniform_continuous_of_tendsto_zero tendsto_integral
```

Comment:
I don't think you need these lemmas to get the continuity. Instead, `norm_integral_le_norm` (that you could move a little bit above) should be enough, together with `linear_map.with_bound`.